### PR TITLE
chore: define GraphQLContext and use it for the GraphQL User type

### PR DESCRIPTION
**Pull request by samerbuna**
@nicolasburtey we can define the context for all resolvers this way

### CODE REVIEW

**Review by nicolasburtey — DISMISSED** [review]

**Comment by nicolasburtey** [comment]
> @nicolasburtey we can define the context for all resolvers this way

This will be very helpful

File: src/graphql/types/object/graphql-user.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import Username from "../scalar/username"
 import * as Users from "@app/users"
 import { UnknownClientError } from "@core/error"
 
-const GraphQLUser = new GT.Object({
+const GraphQLUser = new GT.Object<User, GraphQLContext>({
```
Comment (nicolasburtey) [inline src/graphql/types/object/graphql-user.ts:18]: @samerbuna what should we do got `GT.Field`?

**Review by vindard — APPROVED** [review]

File: src/servers/index.files.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+type GraphQLContext = {
+  logger: Logger
+  uid: UserId | null
+  wallet: Wallet | null
+  domainUser: User | null
+  domainAccount: Account | null
+  user: User | null
+  geetest: GeetestType
+  account: Account | null
+  ip: string | undefined
```
Comment (vindard) [inline src/servers/index.files.d.ts:10]: One nit, we can probably make an `Ip` type for here

Comment (nicolasburtey) [reply src/servers/index.files.d.ts:10]: I think this should be `IpAddress` IIRC

**Pull request merged**